src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -83,15 +92,28 @@ const reducer = (state: InitialState, action: ActionTypes) => {
     case WalletActions.SET_ADAPTERS:
       return { ...state, adapters: action.payload }
     case WalletActions.SET_WALLET:
-      return {
+      const stateData = {
         ...state,
         wallet: action.payload.wallet,
         walletInfo: {
           name: action?.payload?.name,
           icon: action?.payload?.icon,
-          deviceId: action?.payload?.deviceId
+          deviceId: action?.payload?.deviceId,
+          meta: { label: '', address: '' }
         }
       }
+
+      if (action.payload.meta && action.payload.meta.label)
+        stateData.walletInfo.meta.label = action.payload.meta.label
+
+      // ignoring the error here as this prop does not exist on HDWallet but it does
```

Comment:
There shouldn't be a `@ts-ignore` rule here: The type error is because `ethAddress` is specific to the `MetaMaskHDWallet`, `PortisHDWallet` interfaces, but is not present in the more general `HDWallet` one. I just had a quick fiddle, tested this and it works
```javascript
      if ((action.payload.wallet as MetaMaskHDWallet).ethAddress) {
        stateData.walletInfo.meta.address = (action.payload.wallet as MetaMaskHDWallet)?.ethAddress || ''
      }
```

Of course there might be a better way to type it from payload origination without this awful double casting

Reply:
fixed